src/core/router/info.h

```diff
@@ -430,23 +431,33 @@ struct RouterInfoTraits
 class RouterInfo : public RouterInfoTraits, public RoutingDestination
 {
  public:
+  /// @class  AddressInfo
+  /// @brief  Interface Address object to be used by RouterInfo holders.
+  struct AddressInfo
+  {
+    std::string host;
+    std::uint16_t port;
+    // has_transport Supports NTCP, SSU
+    std::pair<bool, bool>
+        has_transport;  // TODO(anonimal): refactor as bitwise SupportedTransport?
+  };
+  using AddressInfoList = std::vector<AddressInfo>;
```

Comment:
> When creating an RI, we should simply pass reference to an Address object or, if we're really picky about memory, something more akin to an AddressBase object (if redesign needs to go that far). An additional perk of using an Address object is that boost::asio::ip::address (as host) can guarantee a valid address syntax sooner than later

Totally agreed.

Now, to put clear expectations, do we want:
* Fully `--host` support + existing `Address` type as part of the RI interface.
* Fully  RI refactoring ( which should be traced through #627 )

being totally 2 different tasks right? I do not want to hold #777 because of #627.

I'll wait for your thoughts about this.
Thanks.

Reply:
>being totally 2 different tasks right?

Yes. Worst case scenario: we get this PR working to resolve #777, write in appropriate TODO's (TODO's regarding what we've discussed here and also point to this PR number), merge the PR, then continue working on #627.

>Now, to put clear expectations, do we want:
>Fully --host support + existing Address type as part of the RI interface.
>Fully RI refactoring ( which should be traced through #627 )

Yes, we want both but, for now, if *you* want: `Fully --host support + existing Address type as part of the RI interface.`. The `Address` class may not fit as perfectly we we'd like, so the outcome remains to be seen. If there is more stumbling, then more PR NACKs may pop-up but I'm open to collaboration if you are (I don't know your patience or interest level yet). If you *don't* want to spend too much time on this PR, then I'd say do the 'worst case scenario' option instead.